tiledb/api/src/array/fragment_info.rs

```diff
@@ -0,0 +1,667 @@
+use std::ops::Deref;
+
+use anyhow::anyhow;
+// use serde::{Deserialize, Serialize};
+// use serde_json::json;
+
+use crate::array::schema::{RawSchema, Schema};
+use crate::config::{Config, RawConfig};
+use crate::context::{CApiInterface, Context, ContextBound};
+use crate::datatype::Datatype;
+use crate::error::Error;
+use crate::fn_typed;
+use crate::key::LookupKey;
+use crate::non_empty_domain::{NonEmptyDomain, Range};
+use crate::string::{RawTDBString, TDBString};
+use crate::Result as TileDBResult;
+
+pub(crate) enum RawFragmentInfo {
+    Owned(*mut ffi::tiledb_fragment_info_t),
+}
+
+impl Deref for RawFragmentInfo {
+    type Target = *mut ffi::tiledb_fragment_info_t;
+    fn deref(&self) -> &Self::Target {
+        let RawFragmentInfo::Owned(ref ffi) = *self;
+        ffi
+    }
+}
+
+impl Drop for RawFragmentInfo {
+    fn drop(&mut self) {
+        let RawFragmentInfo::Owned(ref mut ffi) = *self;
+        unsafe {
+            ffi::tiledb_fragment_info_free(ffi);
+        }
+    }
+}
+
+pub struct FragmentInfo<'ctx> {
+    context: &'ctx Context,
+    raw: RawFragmentInfo,
+}
+
+impl<'ctx> ContextBound<'ctx> for FragmentInfo<'ctx> {
+    fn context(&self) -> &'ctx Context {
+        self.context
+    }
+}
+
+impl<'ctx> FragmentInfo<'ctx> {
+    pub(crate) fn capi(&self) -> *mut ffi::tiledb_fragment_info_t {
+        *self.raw
+    }
+
+    pub fn config(&self) -> TileDBResult<Config> {
+        let mut c_config: *mut ffi::tiledb_config_t = out_ptr!();
+        self.capi_return(unsafe {
+            ffi::tiledb_fragment_info_get_config(
+                self.context.capi(),
+                self.capi(),
+                &mut c_config,
+            )
+        })?;
+
+        Ok(Config::from_raw(RawConfig::Owned(c_config)))
+    }
+
+    pub fn load(&self) -> TileDBResult<()> {
+        self.capi_return(unsafe {
+            ffi::tiledb_fragment_info_load(self.context.capi(), self.capi())
+        })?;
+
+        Ok(())
+    }
+
+    pub fn num_fragments(&self) -> TileDBResult<u32> {
```

Comment:
Our "has a list of stuff" structures are not consistent with the naming convention for getting the number of elements.
FilterList has `get_num_filters`
Domain has `ndim`
Schema has `nattributes`

We probably should make them all consistent.  I'm not sure if there's a convention for this, collections all use `len` which doesn't seem appropriate here.  I'd favor changing them all to `num_thing` or `nthing`

Reply:
FilterList was from before I really internalized the builder pattern, hence the `get_` prefix. My personal preference is for `num_${thing}` but its a lightly held preference. Totes agree that we should normalize one way or the other. I opened [sc-45388] to track the normalization work.